editor/project_settings_editor.cpp

```diff
@@ -144,65 +128,88 @@ void ProjectSettingsEditor::_delete_setting(bool p_confirmed) {
+	_update_property_box();
 }
 
 void ProjectSettingsEditor::_feature_selected(int p_index) {
-	_update_advanced_bar();
+	Vector<String> t = property_box->get_text().strip_edges().split(".", true, 1);
+	const String feature = p_index ? "." + feature_box->get_item_text(p_index) : "";
+	property_box->set_text(t[0] + feature);
+	_update_property_box();
 }
 
-void ProjectSettingsEditor::_update_advanced_bar() {
-	const String property_text = property_box->get_text().strip_edges();
+void ProjectSettingsEditor::_update_property_box() {
+	const String setting = _get_setting_name();
+	const Vector<String> t = setting.split(".", true, 1);
+	const String name = t[0];
+	const String feature = (t.size() == 2) ? t[1] : "";
+	bool feature_invalid = (t.size() == 2) && (t[1] == "");
+
+	add_button->set_disabled(true);
+	del_button->set_disabled(true);
+
+	if (feature != "") {
+		feature_invalid = true;
+		for (int i = 1; i < feature_box->get_item_count(); i++) {
+			if (feature == feature_box->get_item_text(i)) {
+				feature_invalid = false;
+				feature_box->select(i);
+				break;
+			}
+		}
+	}
 
-	String error_msg = "";
-	bool disable_add = true;
-	bool disable_del = true;
+	if (feature == "" || feature_invalid) {
+		feature_box->select(0);
+	}
 
-	if (!property_box->get_text().is_empty()) {
-		const String setting = _get_setting_name();
-		bool setting_exists = ps->has_setting(setting);
-		if (setting_exists) {
-			error_msg = TTR(" - Cannot add already existing setting.");
+	if (property_box->get_text() == "") {
+		return;
+	}
 
-			disable_del = ps->is_builtin_setting(setting);
-			if (disable_del) {
-				String msg = TTR(" - Cannot delete built-in setting.");
-				error_msg += (error_msg == "") ? msg : "\n" + msg;
-			}
+	if (ps->has_setting(setting)) {
+		del_button->set_disabled(ps->is_builtin_setting(setting));
+		_select_type(ps->get_setting(setting).get_type());
+	} else {
+		if (ps->has_setting(name)) {
+			_select_type(ps->get_setting(name).get_type());
 		} else {
-			bool bad_category = false; // Allow empty string.
-			Vector<String> cats = category_box->get_text().strip_edges().split("/");
-			for (int i = 0; i < cats.size(); i++) {
-				if (!cats[i].is_valid_identifier()) {
-					bad_category = true;
-					error_msg = TTR(" - Invalid category name.");
-					break;
-				}
-			}
+			type_box->select(0);
+		}
 
-			disable_add = bad_category;
+		if (feature_invalid) {
+			return;
+		}
 
-			if (!property_text.is_valid_identifier()) {
-				disable_add = true;
-				String msg = TTR(" - Invalid property name.");
-				error_msg += (error_msg == "") ? msg : "\n" + msg;
+		const Vector<String> names = name.split("/");
+		for (int i = 0; i < names.size(); i++) {
+			if (!names[i].is_valid_identifier()) {
+				return;
 			}
 		}
+
+		add_button->set_disabled(false);
 	}
+}
+
+void ProjectSettingsEditor::_select_type(Variant::Type p_type) {
+	int idx = type_box->get_item_index(p_type);
+
+	ERR_FAIL_COND(idx == -1);
```

Comment:
The `select` method already checks if the idx is within selectable options, so the `ERR_FAIL` here is unnecessary.

Reply:
Fixed.